src/ngx_rewrite_driver_factory.cc

```diff
@@ -250,4 +250,23 @@ void NgxRewriteDriverFactory::InitStats(Statistics* statistics) {
   InPlaceResourceRecorder::InitStats(statistics);
 }
 
+void NgxRewriteDriverFactory::PrepareForkedProcess(const char* name) {
+  ngx_pid = ngx_getpid();  // Needed for logging to have the right PIDs.
+  SystemRewriteDriverFactory::PrepareForkedProcess(name);
+}
+
+void NgxRewriteDriverFactory::NameProcess(const char* name) {
+  SystemRewriteDriverFactory::NameProcess(name);
+
+  // Superclass set status with prctl.  Nginx has a helper function for setting
```

Comment:
There are two kinds of process "names": there's the status, 16 chars you can set with `prctl`, and there's `argv[0]`.  Some ways of looking at processes use one, some use the other.  Nginx processes do typically set `argv[0]` so I'd like to do that to fit in.  That's awkward enough [1] that I only want to do it in situations where a helper function is available.

[1] https://cs.corp.google.com/piper///depot/google3/third_party/nginx/src/src/os/unix/ngx_setproctitle.c?q=ngx_setproctitle&l=79&dr=CSs